cpp/simulations/abm.cpp

```diff
@@ -864,20 +866,100 @@ int main()
+        fprintf(f_abm, "%f ", results.get_time(i));
+        auto v = results.get_value(i);
         for (auto j = 0; j < v.size(); ++j) {
             fprintf(f_abm, "%f", v[j]);
             if (j < v.size() - 1) {
                 fprintf(f_abm, " ");
             }
         }
-        if (i < sim.get_result().get_num_time_points() - 1) {
+        if (i < results.get_num_time_points() - 1) {
             fprintf(f_abm, "\n");
         }
     }
     fclose(f_abm);
 }
+
+/**
+ * Run the abm simulation
+ * @param result_dir directory where all results of the parameter study will be stored.
+ * @param num_runs number of runs.
+ * @param save_single_runs [Default: true] Defines if single run results are written to the disk.
+ * @returns any io error that occurs during reading or writing of files.
+ */
+mio::IOResult<void> run(const fs::path& result_dir, size_t num_runs, bool save_single_runs = true)
+{
+
+    auto t0         = mio::abm::TimePoint(0); // Start time (t0) per simulation
+    auto tmax       = mio::abm::TimePoint(0) + mio::abm::days(60); // Endtime (tmax) per simulation
+    auto ensemble_results = std::vector<mio::TimeSeries<double>>{}; // Vector of collected results (TimeSeries objects)
+    ensemble_results.reserve(size_t(num_runs));
+    auto run_idx            = size_t(1); //The run index
+    auto save_result_result = mio::IOResult<void>(mio::success());
+
+    // Loop over a number of run
+    while (run_idx <= num_runs) {
+        // Create the sampled simulation with start time t0
+        auto sim = create_sampled_simulation(t0);
+        // Advance the world to tmax
+        sim.advance(tmax);
+        // Push result of the simulation back to the result vector
+        ensemble_results.push_back(sim.get_result());
+        // Option to save the current run result to file
+        if (save_result_result && save_single_runs) {
+            save_result(sim.get_result(), run_idx, result_dir);
+        }
+        ++run_idx;
+    }
+    BOOST_OUTCOME_TRY(save_result_result);
+    return mio::success();
+}
+
+int main(int argc, char** argv)
+{
+
+    mio::set_log_level(mio::LogLevel::warn);
+
+    std::string result_dir = ".";
+    size_t num_runs = 1;
+    bool save_single_runs = true;
+
+    if (argc == 2) {
+        sscanf(argv[1], "%zu", &num_runs);
+        printf("Number of run is %s.\n", argv[1]);
+        printf("Saving results to the current directory (i.e. build/simulations).\n");
+    } else if (argc == 3) {
+        sscanf(argv[1], "%zu", &num_runs);
+        result_dir = argv[2];
+        printf("Number of run is %s.\n", argv[1]);
+        printf("Saving results to \"%s\".\n", result_dir.c_str());
+    } else {
+        printf("Usage:\n");
+        printf("abm_example <num_runs>\n");
+        printf("\tRun the simulation for <num_runs> time(s).\n");
+        printf("\tStore the results in the current directory (i.e. build/simulations).\n");
+        printf("abm_example <result_dir> <num_runs>\n");
```

Comment:
```suggestion
        printf("abm_example <num_runs> <result_dir>\n");
```

I think it should be inversed here, true?

Reply:
It's done!